app/src/main/kotlin/com/hippo/ehviewer/spider/SpiderQueen.kt

```diff
@@ -455,14 +456,14 @@ class SpiderQueen private constructor(val galleryInfo: GalleryInfo) : CoroutineS
     @Retention(AnnotationRetention.SOURCE)
     annotation class State
     interface OnSpiderListener {
-        fun onGetPages(pages: Int)
-        fun onGet509(index: Int)
-        fun onPageDownload(index: Int, contentLength: Long, receivedSize: Long, bytesRead: Int)
-        fun onPageSuccess(index: Int, finished: Int, downloaded: Int, total: Int)
-        fun onPageFailure(index: Int, error: String?, finished: Int, downloaded: Int, total: Int)
-        fun onFinish(finished: Int, downloaded: Int, total: Int)
-        fun onGetImageSuccess(index: Int, image: Image?)
-        fun onGetImageFailure(index: Int, error: String?)
+        suspend fun onGetPages(pages: Int) {}
+        suspend fun onGet509(index: Int) {}
+        suspend fun onPageDownload(index: Int, contentLength: Long, progressFlow: Flow<Long>) {}
+        suspend fun onPageSuccess(index: Int, finished: Int, downloaded: Int, total: Int) {}
+        suspend fun onPageFailure(index: Int, error: String?, finished: Int, downloaded: Int, total: Int) {}
+        suspend fun onFinish(finished: Int, downloaded: Int, total: Int) {}
+        suspend fun onGetImageSuccess(index: Int, image: Image?) {}
+        suspend fun onGetImageFailure(index: Int, error: String?) {}
```

Comment:
Make `error` and `image` non-null.

Reply:
`error` cannot be non-null unless we refactor `SpiderQueen` later.

Reply:
Done